src/psyclone/domain/lfric/lfric_invoke.py

```diff
@@ -0,0 +1,331 @@
+                    unique_fs.append(fspace)
+                    unique_fs_names.append(fspace.mangled_name)
+        return unique_fs
+
+    @property
+    def operates_on_dofs_only(self):
+        '''
+        :returns: whether or not this Invoke consists only of kernels that \
+                  operate on DoFs.
+        :rtype: bool
+        '''
+        return all(call.iterates_over.lower() == "dof" for call in
+                   self.schedule.kernels())
+
+    def field_on_space(self, func_space):
+        ''' If a field exists on this space for any kernel in this
+        invoke then return that field. Otherwise return None. '''
+        for kern_call in self.schedule.kernels():
+            field = func_space.field_on_space(kern_call.arguments)
+            if field:
+                return field
+        return None
+
+    def gen_code(self, parent):
+        '''
+        Generates Dynamo specific invocation code (the subroutine
+        called by the associated invoke call in the algorithm
+        layer). This consists of the PSy invocation subroutine and the
+        declaration of its arguments.
+        :param parent: The parent node in the AST (of the code to be \
+                       generated) to which the node describing the PSy \
+                       subroutine will be added
+        :type parent: :py:class:`psyclone.f2pygen.ModuleGen`
+
+        '''
+        # Create the subroutine
+        invoke_sub = SubroutineGen(parent, name=self.name,
+                                   args=self.psy_unique_var_names +
+                                   self.stencil.unique_alg_vars +
+                                   self._psy_unique_qr_vars)
+
+        # Declare all quantities required by this PSy routine (invoke)
+        for entities in [self.scalar_args, self.fields, self.lma_ops,
+                         self.stencil, self.meshes,
+                         self.function_spaces, self.dofmaps, self.cma_ops,
+                         self.boundary_conditions, self.evaluators,
+                         self.proxies, self.cell_iterators,
+                         self.reference_element_properties,
+                         self.mesh_properties, self.loop_bounds,
+                         self.run_time_checks]:
+            entities.declarations(invoke_sub)
+
+        # Initialise all quantities required by this PSy routine (invoke)
+
+        if self.schedule.reductions(reprod=True):
+            # We have at least one reproducible reduction so we need
+            # to know the number of OpenMP threads
+            omp_function_name = "omp_get_max_threads"
+            tag = "omp_num_threads"
+            nthreads_name = \
+                self.schedule.symbol_table.lookup_with_tag(tag).name
+            invoke_sub.add(UseGen(invoke_sub, name="omp_lib", only=True,
+                                  funcnames=[omp_function_name]))
+            # Note: There is no assigned kind for integer nthreads as this
```

Comment:
I would suggest single quotations around `'integer'` and `'nthreads'` (language and variable names).